signingscript/src/signingscript/sign.py

```diff
@@ -1633,8 +1633,7 @@ async def apple_notarize(context, path, *args, **kwargs):
     """
     # Setup workdir
     notarization_workdir = os.path.join(context.config["work_dir"], "apple_notarize")
-    shutil.rmtree(notarization_workdir, ignore_errors=True)
-    utils.mkdir(notarization_workdir)
+    utils.mkdir(notarization_workdir, delete_before_create=True)
```

Comment:
This is `delete_before_create` necessary now? Especially for the existing notarization functions?

Reply:
it's more of a safeguard. I don't see why we wouldn't want that..?

Reply:
I guess what I'm wondering is: why is it getting added? Did you run into something? (I don't really have a problem with it...)

Reply:
To be fair, this is more relevant for local testing to ensure we don't have anything in the workdir. AIUI In CI scriptworker takes care of deleting everything after each task is complete.

Reply:
I don't think something called mkdir should be doing rmtree.  Since this seems to be a no-op (we were calling rmtree + mkdir, now we're calling mkdir which itself calls rmtree before re-creating the dir), I'd rather this was reverted.